cypress/integration/my_patches.js

```diff
@@ -32,3 +152,64 @@ describe("My Patches Page", () => {
     });
   });
 });
+
+// TODO: These functions were adopted from cypress/utils/index and fixes two issues
+// which caused the incorrect query to be tracked. The functions in the util folder
+// should be updated as well as any tests that rely on them
+
+const clickingCheckboxUpdatesUrlAndRendersFetchedResults = ({
+  selector = ".cy-checkbox",
+  checkboxDisplayName,
+  pathname,
+  paramName,
+  search,
+  tableRow,
+  query: {
+    name,
+    responseName,
+    checkedRequestVariables,
+    uncheckedRequestVariables,
+  },
+}) => {
+  cy.get(selector)
+    .contains(checkboxDisplayName)
+    .as("target")
+    .click({ force: true });
+  resultsAreFetchedAndRendered({
+    queryName: name,
+    responseName,
+    requestVariables: checkedRequestVariables,
+    tableRow,
+  }).then(() => urlSearchParamsAreUpdated({ pathname, paramName, search }));
+  cy.get("@target").click({ force: true });
+  resultsAreFetchedAndRendered({
+    queryName: name,
+    responseName,
+    requestVariables: uncheckedRequestVariables,
+    tableRow,
+  }).then(() => {
+    urlSearchParamsAreUpdated({ pathname, paramName, search: null });
+  });
+};
+
+const resultsAreFetchedAndRendered = ({
+  queryName,
+  responseName,
+  requestVariables,
+  tableRow,
+} = {}) => {
+  return assertQueryVariables(queryName, requestVariables).then((xhr) => {
+    const { response } = xhr;
+    const numberOfResults = get(response, `body.data.${responseName}`, [])
```

Comment:
Is there a way to more deterministically check that  this data is valid. I think we want to move away from relying on the data we get back from the request and instead comparing the results with expected outcomes.

Reply:
I agree that we should move away from tracking responses from gql and examining responses. We will need to stub our gql queries responses to compare results with expected outcomes. That work is defined in this ticket: https://jira.mongodb.org/browse/EVG-7910

Reply:
we're not stubbing it, it will exist as it is now but we will match against the data in local evergreen instead of what is returned in the response